src/sentry/replays/endpoints/project_replay_details.py

```diff
@@ -44,3 +46,17 @@ def get(self, request: Request, project: Project, replay_id: str) -> Response:
             return Response(status=404)
         else:
             return Response({"data": response[0]}, status=200)
+
+    def delete(self, request: Request, project: Project, replay_id: str) -> Response:
+        if not features.has(
+            "organizations:session-replay", project.organization, actor=request.user
+        ):
+            return Response(status=404)
+
+        recording_segments = ReplayRecordingSegment.objects.filter(
+            project_id=project.id, replay_id=replay_id
+        ).all()
+        for recording_segment in recording_segments:
+            delete_recording_segment.delay(recording_segment.id)
+
+        return Response(status=204)
```

Comment:
should we return a 202 instead of 204 since the deletion isn't immediate? (we return 202 in other pending deletion cases)